qa/L0_simple_ensemble/ensemble_test.py

```diff
@@ -177,6 +178,41 @@ def callback(start_time, result, error):
         if expected_flags != response_flags:
             self.assertTrue(False, "unexpeced sequence flags mismatch error")
 
+    def test_ensemble_partial_add_sub(self):
```

Comment:
I think we should have more testing in general to programatically track expected behavior of certain edge cases, such as:
1. What if a composing model doesn't produce the outputs necessary to feed inputs of the next step?
2. In an ensemble with outputs from 2 parallel steps, what if 1 step has outputs and the other step has no outputs?
3. What if an ensemble doesn't produce any of its outputs?

These aren't immediately clear to me from scanning the code what would happen, so I think it would be good to enforce expected behavior via testing, and be able to catch if the behavior subtlely changes from future ensemble code changes

Reply:
@kthui